src/overlays/actors/ovl_En_Owl/z_en_owl.c

```diff
@@ -31,141 +79,1332 @@ const ActorInit En_Owl_InitVars = {
+            if (gSaveContext.eventChkInf[4] & 1) {
+                // has zelda's letter
+                osSyncPrintf("フクロウ退避\n"); // Owl evacuation
+                Actor_Kill(&this->actor);
+                return;
+            }
+
+            this->actionFunc = EnOwl_WaitKakariko;
+            break;
+        case OWL_HYLIA_GERUDO:
+            if (gSaveContext.eventChkInf[4] & 8) {
+                // has ocarina of time
+                osSyncPrintf("フクロウ退避\n"); // Owl evacuation
+                Actor_Kill(&this->actor);
+                return;
+            }
+            this->actionFunc = EnOwl_WaitGerudo;
+            break;
+        case OWL_LAKE_HYLIA:
+            this->actionFunc = EnOwl_WaitLakeHylia;
+            break;
+        case OWL_ZORA_RIVER:
+            if ((gSaveContext.eventChkInf[3] & 0x200) || !(gSaveContext.eventChkInf[4] & 1)) {
+                // opened zora's domain or has zelda's letter
+                osSyncPrintf("フクロウ退避\n"); // Owl evacuation
+                Actor_Kill(&this->actor);
+                return;
+            }
+
+            this->actionFunc = EnOwl_WaitZoraRiver;
+            break;
+        case OWL_HYLIA_SHORTCUT:
+            this->actionFunc = EnOwl_WaitHyliaShortcut;
+            Flags_UnsetSwitch(globalCtx, 0x23);
+            return;
+        case OWL_DEATH_MOUNTAIN:
+            this->actionFunc = EnOwl_WaitDeathMountainShortcut;
+            break;
+        case OWL_DEATH_MOUNTAIN2:
+            this->actionFunc = EnOwl_WaitDeathMountainShortcut;
+            break;
+        case OWL_DESSERT_COLOSSUS:
+            this->actionFunc = func_80ACB3E0;
+            break;
+        case OWL_LOST_WOODS_PRESARIA:
+            if (!CHECK_QUEST_ITEM(QUEST_SONG_LULLABY)) {
+                osSyncPrintf("フクロウ退避\n"); // Owl evacuation
+                Actor_Kill(&this->actor);
+                return;
+            }
+            this->actionFunc = EnOwl_WaitLWPreSaria;
+            break;
+        case OWL_LOST_WOODS_POSTSARIA:
+            if (!CHECK_QUEST_ITEM(QUEST_SONG_SARIA)) {
+                osSyncPrintf("フクロウ退避\n"); // Owl evacuation
+                Actor_Kill(&this->actor);
+                return;
+            }
+            this->actionFunc = EnOwl_WaitLWPostSaria;
+            break;
+        default:
+            // Outside kokiri forest
+            osSyncPrintf(VT_FGCOL(CYAN));
+            osSyncPrintf("no = %d  \n", owlType);
+            osSyncPrintf(
+                "未完成のフクロウ未完成のフクロウ未完成のフクロウ\n"); // Unfinished owl unfinished owl unfinished owl
+            osSyncPrintf(VT_RST);
+            this->actionFlags |= 2;
+            this->unk_3EE = 0x20;
+            this->actionFunc = EnOwl_WaitOutsideKokiri;
+            break;
+    }
+}
+
+void EnOwl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOwl* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Owl/func_80ACC5CC.s")
+/*
```

Comment:
Ideally, multi-line comments should start with `/**` so they are recognized properly by typical code editors.
Relevant in the rest of the file.
```suggestion
/**
```